labs/src/main/scala/org/scalalabs/basic/lab02/ListManipulationExercise01.scala

```diff
@@ -8,23 +8,20 @@ import sys._
  * An overview of some functional programming techniques can be found here in the O'reilly book:
  * http://programming-scala.labs.oreilly.com/ch08.html#FunctionalDataStructures 
  */
+
+//Modified by Sisi Cheng & Jean Sung
 object ListManipulationExercise01 {
 
   /**
    * Get the first element in the list. Hint: there is a built-in function for this you can use.
    * 
    */
-  def firstElementInList[T](l: List[T]): T = {
-    //buildin
-    null.asInstanceOf[T]
-  }
+  def firstElementInList[T](l: List[T]): T = l.head
 
   /**
    * Get the sum of all the elements in the list, e.g. sumOfList(List(1,2,3)) = 6.
    */
-  def sumOfList(l: List[Int]): Int = {
-    error("fix me")
-  }
+  def sumOfList(l: List[Int]): Int = if (l.isEmpty) 0 else l.head + sumOfList(l.tail)
```

Comment:
The `sum`  function is already provided; I think `tails` is the only function in this file which isn’t.